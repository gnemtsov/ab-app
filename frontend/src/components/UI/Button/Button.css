.Button {
    outline: none;
    cursor: pointer;
    font: inherit;
    color: #fff;
    padding: 0.4rem 0.6rem;
    border-radius: 3px;
    background-color: #449bf7;
    border: none;
}

.Button:disabled {
    background-color: #6c757d;
    opacity: 0.7;
    cursor: not-allowed;
}