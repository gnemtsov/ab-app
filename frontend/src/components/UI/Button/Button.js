import React from 'react';

import classes from './Button.css';

const button = (props) => (
    <button
        disabled={props.disabled}
        className={classes.Button + (props.className ? ' ' + props.className : '')}
        onClick={props.clicked}>
        {props.children}
    </button>
);

export default button;