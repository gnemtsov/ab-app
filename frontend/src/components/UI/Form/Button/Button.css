.Button {
    outline: none;
    cursor: pointer;
    font: inherit;
    color: #000;
    padding: 0.5rem 0.7rem;
    border-radius: 3px;
    background-color: #fcfcfc;
    border-image: none;
    border-style: solid;
    border-width: 1px;
}

.Button:disabled {
    color: #ccc;
    cursor: not-allowed;
}

.Primary {
    color: #fff;
    background-color: #333db3;
    background-image: linear-gradient(#1d2583, #5f6cff);
    border-color: #333db3 #0b4075 #0b4075;
}

.Primary:hover {
    background-image: linear-gradient(#565ca2, #333db3);
    border-color: #333db3 #333db3 #5f6cff;
}

.Danger {
    background-color: #944317;
}