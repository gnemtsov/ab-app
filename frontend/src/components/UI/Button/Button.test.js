import React from 'react';

import { configure, render, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Button from './Button';

configure({ adapter: new Adapter() });

describe('Button', () => {
	it('Renders correctly when disabled', () => {
		const wrapper = shallow(<Button disabled />);
		expect(wrapper).toMatchSnapshot();
	});
	
	it('Renders correctly when not disabled', () => {
		const wrapper = shallow(<Button />);
		expect(wrapper).toMatchSnapshot();
	});
	
	it('Renders correctly with className when disabled', () => {
		const wrapper = shallow(<Button className="test" disabled />);
		expect(wrapper).toMatchSnapshot();
	});
	
	it('Renders correctly with className when not disabled', () => {
		const wrapper = shallow(<Button className="test" />);
		expect(wrapper).toMatchSnapshot();
	});
	
	it('Renders correctly with children', () => {
		const wrapper = shallow(<Button>test</Button>);
		expect(wrapper).toMatchSnapshot();
	});
})