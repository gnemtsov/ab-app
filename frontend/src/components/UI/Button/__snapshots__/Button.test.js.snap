// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders correctly when disabled 1`] = `
<button
  className="Button"
  disabled={true}
/>
`;

exports[`Button Renders correctly when not disabled 1`] = `
<button
  className="Button"
/>
`;

exports[`Button Renders correctly with children 1`] = `
<button
  className="Button"
>
  test
</button>
`;

exports[`Button Renders correctly with className when disabled 1`] = `
<button
  className="Button test"
  disabled={true}
/>
`;

exports[`Button Renders correctly with className when not disabled 1`] = `
<button
  className="Button test"
/>
`;
